### PR TITLE
react-form-controls

### PR DESCRIPTION
**Pull request by dannyandreev**
image

### CODE REVIEW

File: react-form-controls/src/NewsletterForm.jsx [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+
+  render(){
+    return (
+      <form className = "flex">
+        <span>Email</span>
+        <input type="text"
+          onChange={this.handleChange}
+          />
+        <button type="button"
+          onClick={this.handleSubmit}
```
Comment (scbowler) [inline react-form-controls/src/NewsletterForm.jsx:32]: `handleSubmit` should be on the `form` element on the `onSubmit` prop

File: react-form-controls/src/NewsletterForm.jsx [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+    event.preventDefault()
+    console.log(this.state)
+  }
+
+  render(){
+    return (
+      <form className = "flex">
+        <span>Email</span>
+        <input type="text"
+          onChange={this.handleChange}
```
Comment (scbowler) [inline react-form-controls/src/NewsletterForm.jsx:29]: Here you should be setting the `value` prop on the input using `state`

File: react-form-controls/src/NewsletterForm.jsx [hunk 3]
```diff
@@ -0,0 +1,39 @@
+import React from "react"
+
+class NewsletterForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {};
```
Comment (scbowler) [inline react-form-controls/src/NewsletterForm.jsx:6]: Your state object should have starting values for all values in state you are going to use

File: react-form-controls/src/NewsletterForm.jsx [hunk 4]
```diff
@@ -0,0 +1,39 @@
...
+    this.handleClick = this.handleClick.bind(this);
+  }
+
+  handleClick(){
+
+  }
+
```
Comment (scbowler) [inline react-form-controls/src/NewsletterForm.jsx:13]: Don't leave empty methods, if you don't use it, delete it

**Review by scbowler — CHANGES_REQUESTED** [review]
A few updates needed. Let me know if you have any questions

File: react-form-controls/src/NewsletterForm.jsx [hunk 5]
```diff
@@ -0,0 +1,34 @@
+import React from "react"
+
+class NewsletterForm extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {email: ""};
+    this.handleClick = this.handleClick.bind(this);
```
Comment (scbowler) [inline react-form-controls/src/NewsletterForm.jsx:7]: There is no `handleClick` method so this shouldn't be here

File: react-form-controls/src/NewsletterForm.jsx (hunk 2 above)
Comment (scbowler) [reply react-form-controls/src/NewsletterForm.jsx:24]: Still need to add a `value` prop to your input

File: react-form-controls/src/NewsletterForm.jsx [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+
+  render(){
+    return (
+      <form className = "flex">
+        <span>Email</span>
+        <input type="text"
+          onChange={this.handleChange}
+          />
+        <button type="button"
+          onSubmit={this.handleSubmit}
```
Comment (scbowler) [inline react-form-controls/src/NewsletterForm.jsx:27]: `handleSubmit` should be on the `form` elements `onSubmit` prop, not on the `button`

**Review by scbowler — CHANGES_REQUESTED** [review]
A few more updates needed. Let us know if you have any questions

File: react-form-controls/src/NewsletterForm.jsx (hunk 5 above)
Comment (scbowler) [reply react-form-controls/src/NewsletterForm.jsx:7]: Delete this line, `line 7`

File: react-form-controls/src/NewsletterForm.jsx (hunk 6 above)
Comment (scbowler) [reply react-form-controls/src/NewsletterForm.jsx:28]: The `onSubmit` should be on the opening `<form>` tag on line 21

**Review by scbowler — CHANGES_REQUESTED** [review]
A couple more changes needed. If you are unsure on how to fix something please reach out. Also make sure to test your code after making changes.

**Comment by dannyandreev** [comment]
image

**Review by scbowler — CHANGES_REQUESTED** [review]
This one still has the same issues as already noted. Let me know if you have any questions about the requested changes or need any clarification on any of it.

**Review by scbowler — APPROVED** [review]
Great job implementing the feedback, looks good! 👍

**Pull request merged**